src/test/java/org/apache/maven/shared/artifact/filter/AbstractPatternArtifactFilterTest.java

```diff
@@ -579,4 +579,26 @@ public void testmassembly955()
             assertTrue( filter.include( artifact2 ) );
         }
     }
+
+    @Test
+    public void testPartialWildcardShouldNotMatchEmptyComponent()
+    {
+        Artifact artifact = mock( Artifact.class );
+        when( artifact.getGroupId() ).thenReturn( "test-group" );
+        when( artifact.getArtifactId() ).thenReturn( "test-artifact" );
+        when( artifact.getVersion() ).thenReturn( "test-version" );
+        when( artifact.getType() ).thenReturn( null );
```

Comment:
Try test with `DefaultArtifact` implementations - not mock.

https://github.com/apache/maven/blob/master/maven-artifact/src/main/java/org/apache/maven/artifact/DefaultArtifact.java#L126-L151

Reply:
Ok, done -> I had to mock or subclass ArtifactHandler because the classifier was null.

Reply:
I restored your first version with mock, all tests use mocks so do it in the same way